src/lib/certmap/sss_certmap.c

```diff
@@ -914,3 +914,145 @@ void sss_certmap_free_filter_and_domains(char *filter, char **domains)
     talloc_free(filter);
     talloc_free(domains);
 }
+
+static const char *sss_eku_oid2name(const char *oid)
+{
+    size_t c;
+
+    for (c = 0; sss_ext_key_usage[c].name != NULL; c++) {
+        if (strcmp(sss_ext_key_usage[c].oid, oid) == 0) {
+            return sss_ext_key_usage[c].name;
+        }
+    }
+
+    return NULL;
+}
+
+struct parsed_template san_parsed_template[] = {
+    { NULL, NULL, NULL }, /* SAN_OTHER_NAME handled separately */
+    { "subject_rfc822_name", NULL, NULL},
+    { "subject_dns_name", NULL, NULL},
+    { "subject_x400_address", NULL, NULL},
+    { "subject_directory_name", NULL, NULL},
+    { "subject_ediparty_name", NULL, NULL},
+    { "subject_uri", NULL, NULL},
+    { "subject_ip_address", NULL, NULL},
+    { "subject_registered_id", NULL, NULL},
+    { "subject_pkinit_principal", NULL, NULL},
+    { "subject_nt_principal", NULL, NULL},
+    { "subject_principal", NULL, NULL},
+    { NULL, NULL, NULL }, /* SAN_STRING_OTHER_NAME handled separately */
+    { NULL, NULL, NULL }  /* SAN_END */
+};
+
+int sss_cert_dump_content(TALLOC_CTX *mem_ctx, struct sss_cert_content *c,
+                          char **content_str)
+{
+    char *out = NULL;
+    size_t o;
+    struct san_list *s;
+    struct sss_certmap_ctx *ctx = NULL;
+    char *expanded = NULL;
+    int ret;
+    char *b64 = NULL;
+    const char *eku_str = NULL;
+
+    ret = sss_certmap_init(mem_ctx, NULL, NULL, &ctx);
```

Comment:
I know that this function is called just from a single place and its caller as well and the mem_ctx passed by the parent's caller is a tmp_ctx, but I'm not very fond of allocating everything on mem_ctx and never freeing.

Do you think having a tmp_ctx in `sss_certmap_display_cert_content` and adding a comment into this function that it can leave more memory allocated atop mem_ctx would be a good enough change?

btw I'm only nitpicking because this is a library function, if it was in a binary, then I'd say just never free anything..

Reply:
done